staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection.go

```diff
@@ -169,7 +169,7 @@ func (c *connection) newSpdyStream(stream *spdystream.Stream) {
 	err := c.newStreamHandler(stream, replySent)
 	rejectStream := (err != nil)
 	if rejectStream {
-		klog.Warningf("Stream rejected: %v", err)
+		klog.InfoS("Stream rejected", "err", err)
```

Comment:
Here it might make more sense to change the API. This will have implications for code, wo most likely we'll need an alternative API that accepts a logger will keeping the old one around, perhaps with a deprecation notice.

Reply:
The comment is outdated, changed it to contextual logging.
```
-               klog.Warningf("Stream rejected: %v", err)
+               klog.Background().Info("Stream rejected", "err", err)
```

Reply:
`klog.Background().Info` is not contextual logging because it is always using the global logger, regardless of how it got called.

You need to think about ways how a caller can inject its own logger into this function, i.e. change or extend the API.